lib/corefoundation/preferences.rb

```diff
@@ -23,49 +20,115 @@ module CF
   attach_function "CFPreferencesSetValue", %i{key value application_id username hostname}, :void
   attach_function "CFPreferencesAppSynchronize", [:application_id], :bool
 
+  # Interface to the preference utilities from Corefoundation.framework.
+  # Documentation at https://developer.apple.com/documentation/corefoundation/preferences_utilities
+  #
   class Preferences
     CURRENT_USER = CF.kCFPreferencesCurrentUser
     ALL_USERS = CF.kCFPreferencesAnyUser
     CURRENT_HOST = CF.kCFPreferencesCurrentHost
     ALL_HOSTS = CF.kCFPreferencesAnyHost
 
-    class << self
-      def get(key, application_id, username = nil, hostname = nil)
-        # TODO: Check usecase for `CFPreferencesCopyAppValue`
+    # Returns the output from `CFPreferencesCopyValue` call after converting it to ruby type.
+    #
+    # @param [String] key Preference key to read
+    # @param [String] application_id Preference domain for the key
+    # @param [String, Symbol] username Domain user (current, any or a specific user)
+    # @param [String, Symbol] hostname Hostname (current, all hosts or a specific host)
+    #
+    # @return [VALUE] Preference value returned from the `CFPreferencesCopyValue` call.
+    #
+    def self.get(key, application_id, username = nil, hostname = nil)
+      username ||= CURRENT_USER
+      hostname ||= ALL_HOSTS
```

Comment:
current-user and any-host seems to be an appropriate default so I've set this for now. Same defaults are set in `self.set` and `list_keys` methods.